src/components/Menu.tsx

```diff
@@ -1,39 +1,35 @@
 import { Bug, Cog, Files, FlaskConical, GitFork, Monitor, Puzzle, Search, User } from 'lucide-react'
+import type { Icon } from 'lucide-react'
 
 export function Menu() {
   return (
     <div className="flex flex-col justify-between">
       <div className="flex flex-col">
-        <button type="button" className="h-12 flex justify-center items-center border-l-2 border-[#E0DEF2]">
-          <Files strokeWidth={1.5} size={28} color="#E0DEF2" />
-        </button>
-        <button type="button" className="h-12 flex justify-center items-center border-l-2 border-transparent">
-          <Search strokeWidth={1.5} size={28} color="#8F8CA8" />
-        </button>
-        <button type="button" className="h-12 flex justify-center items-center border-l-2 border-transparent">
-          <GitFork strokeWidth={1.5} size={28} color="#8F8CA8" />
-        </button>
-        <button type="button" className="h-12 flex justify-center items-center border-l-2 border-transparent">
-          <Bug strokeWidth={1.5} size={28} color="#8F8CA8" />
-        </button>
-        <button type="button" className="h-12 flex justify-center items-center border-l-2 border-transparent">
-          <Puzzle strokeWidth={1.5} size={28} color="#8F8CA8" />
-        </button>
-        <button type="button" className="h-12 flex justify-center items-center border-l-2 border-transparent">
-          <Monitor strokeWidth={1.5} size={28} color="#8F8CA8" />
-        </button>
-        <button type="button" className="h-12 flex justify-center items-center border-l-2 border-transparent">
-          <FlaskConical strokeWidth={1.5} size={28} color="#8F8CA8" />  
-        </button>
+        <Button Icon={Files} isActive />
+        <Button Icon={Search} />
+        <Button Icon={GitFork} />
+        <Button Icon={Bug} />
+        <Button Icon={Puzzle} />
+        <Button Icon={Monitor} />
+        <Button Icon={FlaskConical} />
       </div>
       <div className="flex flex-col items-center">
-        <button type="button" className="h-12 flex justify-center items-center border-l-2 border-transparent">
-          <User strokeWidth={1.5} size={28} color="#8F8CA8" />
-        </button>
-        <button type="button" className="h-12 flex justify-center items-center border-l-2 border-transparent">
-          <Cog strokeWidth={1.5} size={28} color="#8F8CA8" />
-        </button>
+        <Button Icon={User} />
+        <Button Icon={Cog} />
       </div>
     </div>
   )
+}
+
+interface ButtonProps {
+  Icon: Icon
```

Comment:
Is there any reason to use a capital `I` on the Icon property? Can we keep it lowercase?

Reply:
Yes there were a reason, React wouldn't allow me to create a new icon component if I passed the parameter in lowercase, so `<icon />` was throwing errors. I did a small workaround with the parameter, so now there will be lowercase usage with this component but its function will create an `<Icon />` in uppercase. Please let me know what you think

Reply:
Got it, you could also rename the parameter inside the destrutcturing assignment, some like 
```tsx
function MenuButton({ icon: Icon, isActive = false }: MenuButtonProps) {
```
what do you think?